### PR TITLE
14425-installation-fails-when-user-or-database-already-exists

### PR DESCRIPTION
**Pull request by frolui**

### CODE REVIEW

**Review by Zkir — COMMENTED** [review]
Looks good for me, but my level of psql understanding is low.
@lgudyma-kontur any comments?

File: scripts/Makefile [hunk 1]
```diff
@@ -71,11 +71,19 @@ ${GEOCINT_WORK_DIRECTORY}/geocint/installation/installation_block_finished: ${GE
 # ------------------------ DB configuration block ----------------------------
 
 ${GEOCINT_WORK_DIRECTORY}/geocint/installation/create_user: ${GEOCINT_WORK_DIRECTORY}/geocint/installation/installation_block_finished ## create postgresql user
-	sudo -u postgres psql -c "CREATE ROLE ${PGUSER} WITH SUPERUSER LOGIN ENCRYPTED PASSWORD '${PGUSER}pass';"
+	rm -f USER_COUNT
+	sudo -u postgres psql -q -X -t -c "select count(*) from pg_roles WHERE rolname = '${PGUSER}'" | xargs echo > USER_COUNT
+	if [ $$(cat USER_COUNT) -lt 1 ]; then sudo -u postgres psql -c "CREATE ROLE ${PGUSER} WITH SUPERUSER LOGIN ENCRYPTED PASSWORD '${PGUSER}pass';"; fi
```
[the remark this answers is not in the document]
Comment (frolui) [reply scripts/Makefile]: replaced with PGPASSWORD variable from config.inc.sh file

File: scripts/Makefile [hunk 2]
```diff
@@ -71,11 +71,19 @@ ${GEOCINT_WORK_DIRECTORY}/geocint/installation/installation_block_finished: ${GE
...
+	if [ 0 -lt $$(cat USER_COUNT) ]; then echo "role ${PGUSER} alredy exists. Skip creation."; fi
```
[the remark this answers is not in the document]
Comment (frolui) [reply scripts/Makefile]: fixed

File: scripts/Makefile [hunk 3]
```diff
@@ -71,11 +71,19 @@ ${GEOCINT_WORK_DIRECTORY}/geocint/installation/installation_block_finished: ${GE
...
+	rm -f USER_COUNT
 	touch $@
 	
 ${GEOCINT_WORK_DIRECTORY}/geocint/installation/create_database: ${GEOCINT_WORK_DIRECTORY}/geocint/installation/create_user ## create postgresql db
-	sudo -u postgres psql -c "CREATE DATABASE ${PGDATABASE} WITH OWNER ${PGUSER};"
+	rm -f DB_COUNT
+	sudo -u postgres psql -q -X -t -c "select count(*) FROM pg_database WHERE datname = '${PGDATABASE}'" | xargs echo > DB_COUNT
+	if [ $$(cat DB_COUNT) -lt 1 ]; then sudo -u postgres psql -c "CREATE DATABASE ${PGDATABASE} WITH OWNER ${PGDATABASE};"; fi
```
[the remark this answers is not in the document]
Comment (frolui) [reply scripts/Makefile]: fixed

File: scripts/Makefile [hunk 4]
```diff
@@ -71,11 +71,19 @@ ${GEOCINT_WORK_DIRECTORY}/geocint/installation/installation_block_finished: ${GE
...
+	if [ 0 -lt $$(cat DB_COUNT) ]; then echo "database ${PGDATABASE} alredy exists. Skip creation."; fi
```
[the remark this answers is not in the document]
Comment (frolui) [reply scripts/Makefile]: fixed

**Review by lgudyma-kontur — APPROVED** [review]

**Review by Zkir — APPROVED** [review]

**Pull request merged**